### ISSUE TITLE
Refactor spawn system to be time based

### ISSUE THREAD

**Issue by jjonj** [Revolutionary-Games/Thrive#124]
The spawn system seems to be acting a bit wierd with regards to AI entities. It doesn't seem to take number of existing entities into account, which can cause a serious buildup of AI entities around the player and possibly other unwanted effects.

**Comment by hhyyrylainen** [comment]
The spawn system should be edited to take into account the number of currently spawned entities of each type. There should also be limits per type like how many floating organelles or cells of each species (perhaps depending on the population) can be spawned.

**Comment by Untrustedlife** [comment]
This seems fun, i might attack this after i get player shooting toxins working (though, it is a first issue, so maybe i should leave it for someone new and work on other stuff, though this came up recently)

**Comment by Untrustedlife** [comment]
I think i might tackle this. Unless a newer programmer isnt afraid to take it on.

**Comment by hhyyrylainen** [comment]
Maybe related: https://community.revolutionarygamesstudio.com/t/lag-number-of-enemy-live-forms-rises-to-infinity/1497?u=hhyyrylainen

**Comment by hhyyrylainen** [comment]
For the godot version: verify that spawn system works nicely and then delete spawn_system.cpp

**Comment by hhyyrylainen** [comment]
Clouds might now be rarer than intended.

**Comment by hhyyrylainen** [comment]
Basically at this point I think that the spawn system should be refactored to be time based instead of player movement based, it'll be much better that way.

**Comment by CoreyHendrey** [comment]
I'll discuss this in here due to timezones making it hard to discuss on Discord...

Is there more information on what we want to achieve other than just "time based"?

Is there going to be a set interval where the spawn algorithm runs? If so, this can pretty much be done with just the removal of the check for the player moving. Though that seems suboptimal.

I was thinking maybe we assess how many of each species has been spawned in a certain radius around the player currently, and keep running the SpawnSystem in intervals until each species has reached their maximum amount spawned (or total entity cap is reached).

I problem with this at the moment is the clouds. They are not added to the entity count currently, and as such will spawn infinitely if the player stands still. But I guess we could just add a cap to those as well.

**Comment by hhyyrylainen** [comment]
Here's my random thoughts on how the spawn system should work to solve a bunch of gameplay issues:
## Microbe Spawns
Basically I think that the spawn system should spawn a random cell every `n` seconds with a weighted distribution based on the fraction that species has of the total population of all the species.
There should be a limit for each species based on its population so that if there are just species with low populations then there isn't a lot of cells. But when the populations are high the cap should be added to keep the game from totally lagging.
In addition to the time based spawning, there should be some additional stuff spawned when the player moves around.
Also I think that things should mostly spawn in the direction the player is generally moving in (just like before). So when stationary things would still kinda roughly spawn in the direction the player was going, but there would also be added randomness to spawn some things also behind the player.


## Chunk spawns
the same logic as microbe spawns, except maybe these could be separately capped to not take up the entity limit that microbes take up, also when stationary the chunk spawn rate should be greatly reduced, so moving would spawn more chunks. This would be different from microbes that would still have a relatively high spawn rate when the player is stationary.

## Compound clouds
Basically the same as the other two, spawn a few clouds in a time based manner, and spawn more clouds when the player is moving. The spawn rate probably has to be quite low when stationary as it now takes quite a while for a cloud to dissipate.

Also one general thing, currently the clouds can spawn on screen, which doesn't look that good, so the radius at which clouds spawn should be a bit increased.

I wrote this in one go so this might not make sense entirely.

**Comment by hhyyrylainen** [comment]
Fixing this will also fix: #1264
